### ISSUE TITLE
sh "pwd" returning ./fastlane directory is somewhat misleading

### ISSUE THREAD

**Issue by aafa** [KrauseFx/fastlane#155]
I've found that my shell actions beginning with "cd.." is a very common case
```
sh "cd ..; my_script"
```

but at the same time other parts have their current directories as I expect -- from project root

```
  xcexport(
      archive_path: "./fastlane/latest.xcarchive",
      export_path: "./fastlane/latest"
    )
```

I think it'd be better to have more consistency here one way or another.

**Comment by KrauseFx** [comment]
Where do you need to do `..`? In the `Deliverfile` and `Snapfile` I guess? Also in new `fastlane` actions?

**Comment by aafa** [comment]
Fastfile and Deliverfile is number one candidates to me.  But still it would be nice to have single base dir  everywhere.

**Comment by KrauseFx** [comment]
I agree, there were some mix-ups with the introduction of `fastlane` after the existing tools were already out there.

**Comment by KrauseFx** [comment]
But changing this now would cause existing configuration files to break, which is probably not something we want.

**Comment by aafa** [comment]
I agree, no one would be happy to see their configs broken :) I've been thinking on maybe introduce special ENV variable with base directory, which be defaulted with current one.  But on the other hand that will add additional unnecessary complexity.  

So the only plausible solution here, I think, is to emphasize in docs this fact about "sh" - action.

**Comment by KrauseFx** [comment]
I guess this could be changed with a major version release sometime in the future, but we should leave it as it is for now. Sorry about the troubles, I'm not super happy with it as well.

**Comment by johnboiles** [comment]
(posting here because this issue was good SEO)

Looks like this changed with Fastlane 1.33.1 (and perhaps earlier). My `sh` commands are now executed  from the directory `fastlane` is executed in, instead of the directory that contains the Fastfile.

This change did break my configuration, but It's probably the right thing to do.

**Comment by KrauseFx** [comment]
@johnboiles I didn't change anything in that regard. I think the fastlane actions run in a different context depending on if it's inside a lane or outside.

**Comment by johnboiles** [comment]
@KrauseFx you're right, I should have dug further before posting. The issue on this thread isn't quite the same. Looks like the culprit in my case is the line I was using this to get the directory of the Fastfile:

```
ENV["FASTFILE_DIR"] =  File.expand_path(File.dirname(File.dirname(__FILE__)))
```

Something changed (perhaps with El Capitan?) that causes this line to no longer do the same thing as it used to. `puts __FILE__` in the Fastfile gives `../fastlane/Fastfile`. Seems strange to me that `__FILE__` would be prefixed by `..` -- I don't think it had that in the past. At any rate, this is unlikely to be a Fastlane issue.

**Comment by KrauseFx** [comment]
@johnboiles Maybe this is related to [this pull request](https://github.com/KrauseFx/fastlane/pull/683/files) via @samnung?

**Comment by samnung** [comment]
It could be my mistake in that pull request, but I changed only path to eval function, documentation is saying that it is used only for exception backtrace, nothing else (see http://ruby-doc.org/core-2.0.0/Kernel.html#method-i-eval). Or am I wrong?

When I add `puts __FILE__` to my Fastfile, it prints this: `[08:55:30]: Fastfile`. Well that's strange, it should be absolute path to Fastfile or path from root of the iOS project, not from `fastlane` folder.

**Comment by johnboiles** [comment]
I did a little more digging and this seems to be an issue with the casing of the `fastlane` directory. I created the directory as `Fastlane` and not `fastlane`. To test, I created a simple `Fastfile`:

```ruby
fastlane_version "1.3.2"
lane :dir_test do
  puts __FILE__
  puts File.expand_path(File.dirname(File.dirname(__FILE__)))
end
```
If I put this file under `Fastlane/Fastfile` I get the following (removing uninteresting lines)
```
[10:18:50]: Driving the lane 'dir_test' 🚀
[10:18:50]: ../fastlane/Fastfile
[10:18:50]: /Users/johnboiles/fastlane_test
```
However if I put the file under `fastlane/Fastfile`
```
[10:18:25]: Driving the lane 'dir_test' 🚀
[10:18:25]: Fastfile
[10:18:25]: /Users/johnboiles/fastlane_test/fastlane
```
@samnung's PR does seem like a likely candidate for causing this issue. Perhaps the `relative_path = path.nil? ? '(eval)' : Pathname.new(path).relative_path_from(Pathname.new(Dir.pwd)).to_s` line is doing something expected.

**Comment by johnboiles** [comment]
@KrauseFx your call as to whether this is worth fixing or not. Naming the dir Fastlane vs fastlane is a purely aesthetic issue. (FWIW capitals seem more consistent with typical Cocoa directory naming :smile: )

I suspect there's some issue related to hardcoding the `fastlane` dir https://github.com/KrauseFx/fastlane/blob/5fcdcac3b66083f3fdbf17ddfa4d8b6ae0673595/lib/fastlane/fastlane_folder.rb#L3

**Comment by samnung** [comment]
@johnboiles @KrauseFx Ok, probably that's the problem, I tried to fix this in eb193c7154d2b34cfef292f26db0faf63ddbcefd. What do you think about it?

**Comment by johnboiles** [comment]
@samnung thanks for the fix! Looks sensible to me, but I'll defer to @KrauseFx as my Ruby isn't great.
